psi4/src/psi4/fisapt/fisapt.cc

```diff
@@ -525,7 +525,7 @@ void FISAPT::nuclear() {
 
     auto Vfact = std::make_shared<IntegralFactory>(primary_);
     std::shared_ptr<PotentialInt> Vint;
-    Vint = std::shared_ptr<PotentialInt>(static_cast<PotentialInt*>(Vfact->ao_potential()));
+    Vint = std::shared_ptr<PotentialInt>(static_cast<PotentialInt*>(Vfact->ao_potential().release()));
```

Comment:
Does `Vint = Vfact->ao_potential()` not suffice here (and many other places throughout this PR)?

Reply:
Not without changing `std::shared_ptr<PotentialInt> Vint;` on L527, Im not libmints expert but does this have downstream effects or are we all good to cast to PotentialInt?

Reply:
Lots of cases like this in this PR so probably good to decide what the strategy is.

Reply:
To be more explicit: this line of code is both downcasting the underlying object and changing the type of the holding smart pointer to a shared pointer. I missed that first point in my original comment.

The underlying `ao_potential` method should be changed so that this downcasting isn't necessary, but that's beyond scope for this PR. This can stay as-is, begrudgingly, knowing that the removal of this ugliness is deferred to a follow-up PR. I'll do it if need be.

Reply:
I see. This is in the same vein as #2793 and related PR #2795.  Perhaps should make a more general issue. Happy to help out fixing.